homeassistant/components/epson/config_flow.py

```diff
@@ -0,0 +1,115 @@
+"""Config flow for epson integration."""
+import epson_projector as epson
+from epson_projector.const import POWER
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_SSL,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, TIMEOUT_SCALE
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_NAME, default=DOMAIN): str,
+        vol.Required(CONF_PORT, default=80): int,
+        vol.Required(CONF_SSL, default=False): bool,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    epson_proj = epson.Projector(
+        data[CONF_HOST],
+        websession=async_get_clientsession(hass, verify_ssl=data.get(CONF_SSL, False)),
+        port=data[CONF_PORT],
+    )
+    _power = await epson_proj.get_property(POWER)
+    if not _power or _power == STATE_UNAVAILABLE:
+        raise CannotConnect
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for epson."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Epson options flow."""
+        return EpsonOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            if self.host_already_configured(user_input[CONF_HOST]):
+                return self.async_abort(reason="already_configured")
```

Comment:
We should use unique ID. Once you have the input validated, call `await self.async_set_unique_id(unique_id)`. Fetch the unique ID when you validate the input.